tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -785,6 +785,30 @@ final case class TestingServer(
     server.didChangeConfiguration(params).asScala
   }
 
+  def willRenameFiles(
+      files: Set[String],
+      fileRenames: Map[String, String],
+  ): Future[Map[String, String]] = {
+    val renames = fileRenames.toList.map { case (oldFilename, newFilename) =>
+      val oldUri = workspace.resolve(oldFilename).toURI.toString
+      val newUri = workspace.resolve(newFilename).toURI.toString
+      new l.FileRename(oldUri, newUri)
+    }.asJava
+    val params = new l.RenameFilesParams(renames)
+    for {
+      editsOrNull <- server.willRenameFiles(params).asScala
+      edits = Option(editsOrNull).getOrElse(new WorkspaceEdit)
+      updatedSources = files.map { file =>
```

Comment:
```suggestion
      updatedSources = fileRenames.keySet.map { file =>
```
would not work?

Reply:
Hm
The logic here is to 
- calculate workspace edits triggered by changes in a few file names/locations. These changes are captured in `fileRenames`
- apply edits to each file in the wokspace. `files` here (renamed to `workspaceFiles`)  contain names of these files

So, here we want to iterate over every file, not only over a renamed ones

Reply:
Ok, make sense!